pkg/config/config.go

```diff
@@ -178,17 +186,20 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		if _, err := url.Parse(rc.APIURL); err != nil {
 			return fmt.Errorf("invalid api_url %q in receiver %q: %s", rc.APIURL, rc.Name, err)
 		}
-		if rc.User == "" {
-			if c.Defaults.User == "" {
-				return fmt.Errorf("missing user in receiver %q", rc.Name)
-			}
-			rc.User = c.Defaults.User
+
+		if rc.User != "" && rc.Password != "" && rc.PersonalAccessToken != "" {
+			return fmt.Errorf("bad auth config in receiver %q: user/password and PAT authentication are mutually exclusive", rc.Name)
 		}
-		if rc.Password == "" {
-			if c.Defaults.Password == "" {
-				return fmt.Errorf("missing password in receiver %q", rc.Name)
+
+		if (rc.User == "" || rc.Password == "") && rc.PersonalAccessToken == "" {
+			if c.Defaults.User != "" && c.Defaults.Password != "" {
+				rc.User = c.Defaults.User
+				rc.Password = c.Defaults.Password
+			} else if c.Defaults.PersonalAccessToken != "" {
```

Comment:
hmm, why not setting those defaults one by one, without caring about dependencies from each?

Reply:
Same thing as in https://github.com/prometheus-community/jiralert/pull/110#discussion_r825512228.

Reply:
resolving in favour of https://github.com/prometheus-community/jiralert/pull/110#discussion_r825512228